sources/api/thar-be-updates/src/main.rs

```diff
@@ -277,7 +280,11 @@ async fn drive_state_machine(
         }
         // Refreshing the list of updates is allowed under every update state
         (UpdateCommand::Refresh, _) => {
-            refresh(update_status, socket_path).await?;
+            // scope for the tokio runtime to this block
```

Comment:
This is one thing that bugs me about this option. You have to be really careful about the scoping, or you could create an extra runtime and panic. It's also uncommon to create runtimes in multiple places. If we were going to create it, I think it might make more sense to make sure we always have one, by using the normal tokio::main annotation and then creating one when we fork. It requires awkward wrapping, but only in one place... (I'm hoping we could come up with something less awkward later, just using the simple reordering solution for now...)

Reply:
> then creating one when we fork

I don't quite understand that. Do you have to also shut-down the existing runtime before the call to fork?

Reply:
I think it'd effectively disappear, and you just start a new one in the forked process like it was the first time.